### PR TITLE
add barista router

### PR DESCRIPTION
**Pull request by kieran**
Nice work on fmw - easily trounces barista :-)

I left the benchmarks for you to update

### CODE REVIEW

**Review by delvedor — CHANGES_REQUESTED** [review]
Hi! Thank you for contributing!
Can you please remove the lockfile? (and maybe add it to the gitignore)

File: README.md [hunk 1]
```diff
@@ -53,7 +54,7 @@ express | lookup wildcard route x 48,479 ops/sec ±109.54% (16 runs sampled)
 ```
 
 ### Run the benchmarks
-Do you wan to run the benchmarks by yourself?  
+Do you wan to run the benchmarks by yourself?
```
Comment (delvedor) [inline README.md]: Can you restore this?

**Comment by kieran** [comment]
updated

**Review by delvedor — APPROVED** [review]
Thanks!

**Comment by delvedor** [comment]
Did you try run the benchmarks with `npm run all` on you machine?
I got this:
```
<--- Last few GCs --->

   50250 ms: Mark-sweep 1382.7 (1438.1) -> 1382.7 (1438.1) MB, 1321.8 / 0.0 ms [allocation failure] [GC in old space requested].
   51649 ms: Mark-sweep 1382.7 (1438.1) -> 1381.9 (1420.1) MB, 1398.1 / 0.0 ms [last resort gc].
   53026 ms: Mark-sweep 1381.9 (1420.1) -> 1383.2 (1420.1) MB, 1377.3 / 0.0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0xd7e9a63fa99 <JS Object>
    1: match [/***/router-benchmark/node_modules/express/lib/router/layer.js:~110] [pc=0xcd56c9cbefa] (this=0x8c195723d79 <a Layer with map 0x3ad237d9a3c9>,path=0x2a59ce1c7bd9 <String[11]: /bench/mark>)
    2: matchLayer(aka matchLayer) [/***/router-benchmark/node_modules/express/lib/router/index.js:574] [pc=0xcd56c9c7533] (this=0xd7e9a604241 <undefi...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/***/.nvm/versions/node/v6.11.1/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/***/.nvm/versions/node/v6.11.1/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/***/.nvm/versions/node/v6.11.1/bin/node]
 4: v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [/***/.nvm/versions/node/v6.11.1/bin/node]
 5: v8::internal::Runtime_AllocateInTargetSpace(int, v8::internal::Object**, v8::internal::Isolate*) [/***/.nvm/versions/node/v6.11.1/bin/node]
 6: 0xcd56c9060c7
 7: 0xcd56c9cbefa
 8: 0xcd56c9c7533
 9: 0xcd56c9c6c8e
sh: line 1: 19521 Abort trap: 6           npm run static
```

**Comment by kieran** [comment]
worked for me! Not terribly flattering, but it ran.

```
$ npm run all

> benchrouter@1.0.0 all /Users/kieran/work/router-benchmark
> npm run static && npm run dynamic && npm run wildcard


> benchrouter@1.0.0 static /Users/kieran/work/router-benchmark
> node static.js

find-my-way | lookup static route x 5,350,543 ops/sec ±1.17% (85 runs sampled)
route-recognizer | lookup static route x 187,273 ops/sec ±0.80% (85 runs sampled)
router | lookup static route x 135,878 ops/sec ±7.78% (52 runs sampled)
server-router | lookup static route x 107,312 ops/sec ±4.00% (58 runs sampled)
koa-router | lookup static route x 164,636 ops/sec ±5.47% (63 runs sampled)
routr | lookup static route x 728,624 ops/sec ±6.14% (68 runs sampled)
call | lookup static route x 6,416 ops/sec ±3.62% (60 runs sampled)
express | lookup static route x 125,596 ops/sec ±26.99% (37 runs sampled)
barista | lookup static route x 58,286 ops/sec ±72.61% (41 runs sampled)

> benchrouter@1.0.0 dynamic /Users/kieran/work/router-benchmark
> node dynamic.js

find-my-way | lookup dynamic route x 1,321,948 ops/sec ±1.06% (84 runs sampled)
route-recognizer | lookup dynamic route x 133,407 ops/sec ±0.89% (87 runs sampled)
router | lookup dynamic route x 101,347 ops/sec ±8.79% (67 runs sampled)
server-router | lookup dynamic route x 100,774 ops/sec ±3.56% (68 runs sampled)
koa-router | lookup dynamic route x 188,463 ops/sec ±4.17% (71 runs sampled)
routr | lookup dynamic route x 355,804 ops/sec ±2.81% (66 runs sampled)
call | lookup dynamic route x 7,305 ops/sec ±3.39% (66 runs sampled)
express | lookup dynamic route x 92,258 ops/sec ±35.62% (64 runs sampled)
barista | lookup dynamic route x 56,867 ops/sec ±5.19% (70 runs sampled)

> benchrouter@1.0.0 wildcard /Users/kieran/work/router-benchmark
> node wildcard.js

find-my-way | lookup wildcard route x 1,498,468 ops/sec ±1.33% (82 runs sampled)
route-recognizer | lookup wildcard route x 185,960 ops/sec ±0.87% (87 runs sampled)
router | lookup wildcard route x 108,941 ops/sec ±7.56% (63 runs sampled)
server-router | lookup wildcard route x 95,391 ops/sec ±4.03% (64 runs sampled)
koa-router | lookup wildcard route x 179,115 ops/sec ±4.64% (69 runs sampled)
routr | lookup wildcard route x 360,845 ops/sec ±4.18% (56 runs sampled)
express | lookup wildcard route x 94,257 ops/sec ±33.68% (54 runs sampled)
barista | lookup wildcard route x 62,865 ops/sec ±5.07% (68 runs sampled)
```

**Pull request merged**